### DISCUSSION TITLE
self.local_rank in LightningDataModule

### DISCUSSION THREAD

**Discussion by austinmw** [Lightning-AI/pytorch-lightning#12056] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Hi, I was recently reading this example from NVIDIA DALI: 
https://github.com/NVIDIA/DALI/blob/629c57592b9b4e91b8213e6c77c1af179f7dd079/docs/examples/frameworks/pytorch/pytorch-lightning.ipynb

I wanted to split the model and datamodule apart. In that case, how can I get `local_rank`, `global_rank` and `world_size` for datamodule's setup?

**Comment by rohitgr7** [comment, marked as answer]
hey @austinmw !

you can access them using `self.trainer.local_rank`, `self.trainer.global_rank` & `self.trainer.world_size`.

**Reply by austinmw** [reply]
Thanks!
